app/routes.py

```diff
@@ -1 +1,208 @@
-from flask import Blueprint
\ No newline at end of file
+import requests
+from flask import Blueprint, jsonify, request, abort, make_response 
+from app import db 
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+import json
+
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+@tasks_bp.route("", methods=["POST"])
+def add_task():
+    request_body = request.get_json()
+
+    if "description" not in request_body\
+        or "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(title=request_body["title"], description=request_body["description"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify(new_task.to_dict()), 201
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    response = []
+    sort_query = request.args.get("sort")
+    if sort_query is None:
+        all_tasks = Task.query.all()
+
+    elif sort_query == "asc":
+        all_tasks = Task.query.order_by(Task.title.asc())
+    else:
+        all_tasks = Task.query.order_by(Task.title.desc())
+    
+    for task in all_tasks:
+        response.append(task.to_dict()["task"])
+    
+    return jsonify(response), 200
+
+@tasks_bp.route("<task_id>", methods=["GET"])
+def get_task_by_id(task_id):
+    task = Task.query.get_or_404(task_id)
+    response = task.to_dict()
+    return jsonify(response), 200
+
+@tasks_bp.route("<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    request_data = request.get_json()
+
+    task.title = request_data["title"]
+    task.description = request_data["description"]
```

Comment:
I'd recommend wrapping these lines in a try-except block so that you can return an error message if the user is missing some of the required data